.github/workflows/main-pr.yaml

```
@@ -0,0 +1,90 @@
+          # Fetch tags from the main branch
+          git fetch origin main --tags
+          # Get the latest tag from the main branch
+          TAG=$(git tag --list --sort=-v:refname | grep -E '^v[0-9]+\.[0-9]+\.[0-9]+' | head -n 1 || echo "0.0.0")
+          # Remove the 'v' prefix from the tag
+          CLEAN_TAG=$(echo $TAG | sed 's/^v//')
+          echo "LATEST_TAG=$CLEAN_TAG" >> $GITHUB_ENV
+          echo "LATEST_TAG=$CLEAN_TAG" > $GITHUB_ENV
+
+      - name: Compare versions
+        id: compare_versions
+        run: |
```

actionlint

shellcheck reported issue in this script: SC2086:info:17:29: Double quote to prevent globbing and word splitting

(shellcheck)